test-validator/src/lib.rs

```diff
@@ -790,12 +793,31 @@ impl TestValidator {
         let validator_stake_lamports = sol_to_lamports(1_000_000.);
         let mint_lamports = sol_to_lamports(500_000_000.);
 
+        let mut feature_set = FeatureSet::all_enabled();
+        for feature in &config.deactivate_feature_set {
+            feature_set.deactivate(feature);
+        }
```

Comment:
This duplicates a similar logic below:

https://github.com/anza-xyz/agave/blob/3deac722ded065f47b36936d27480319f55dc029/test-validator/src/lib.rs#L859-L870

I wanted to avoid making changes to the existing code to make it easier to review. Would it be better to refactor this to avoid duplicating the logic, or should I keep it as is?